syntax = "proto3";

package sumparse;

message HashSum {
  enum Type {
    TYPE_UNKNOWN = 0;
    TYPE_MD5 = 1;
    TYPE_SHA1 = 2;
  }

  message File {
    enum Type {
      TYPE_UNKNOWN = 0;  // should never happen
      TYPE_BINARY = 1;
      TYPE_TEXT = 2;
    }

    Type type = 1;
    string name = 2;
    string hash = 3;
  }

  Type type = 1;
  repeated File file = 2;
}
